src/core/net/ip6.cpp

```diff
@@ -1037,6 +1037,19 @@ Error Ip6::PassToHost(OwnedPtr<Message> &aMessagePtr,
     }
 #endif
 
+#if OPENTHREAD_CONFIG_ANDROID_AOSP_ENABLE
```

Comment:
The change looks good.

I would suggest using a different `CONFIG` name that describes the behavior it enable, instead of `ANDRIOD`.
- We want to keep the OT stack independent of underlying OS and platform.
- The same config/behavior may be useful on other platforms that are not based on Andriod.

Some suggestions: 
- `OPENTHREAD_CONFIG_IP6_RESTRICT_FORWARDING_LARGER_SCOPE_MCAST_WITH_LOCAL_SRC`
- `OPENTHREAD_CONFIG_IP6_SET_HOP_LIMIT_TO_ONE_FOR_MCAST_WITH_ML_LL_SRC`
- ...?

Reply:
Thanks, sounds good to have the config name describing the behavior so it can be reused by any platform when needed.  `OPENTHREAD_CONFIG_IP6_RESTRICT_FORWARDING_LARGER_SCOPE_MCAST_WITH_LOCAL_SRC` looks good, will update to it.